src/main/kotlin/cache/loaders/TextureLoader.kt

```diff
@@ -49,35 +56,95 @@ class TextureLoader(cacheLibrary: CacheLibrary) {
         return textureDefinitionCache[id]
     }
 
-    private fun load(id: Int, b: ByteArray?): TextureDefinition {
-        val def = TextureDefinition()
-        val inputStream = ByteBuffer.wrap(b)
-        def.field1777 = inputStream.readUnsignedShort()
-        def.field1778 = inputStream.get().toInt() != 0
-        def.id = id
-        val count: Int = inputStream.readUnsignedByte()
-        val files = IntArray(count)
-        for (i in 0 until count) files[i] = inputStream.readUnsignedShort()
-        def.fileIds = files
-        if (count > 1) {
-            def.field1780 = IntArray(count - 1)
-            for (var3 in 0 until count - 1) {
-                def.field1780[var3] = inputStream.readUnsignedByte()
-            }
+    private fun load(id: Int, b: ByteArray?): TextureDefinition? {
+        if (b == null || b.isEmpty()) {
```

Comment:
Why do we want to allow null/empty byte arrays to be passed to this function in the first place?